includes/apple-exporter/components/class-pre.php

```diff
@@ -0,0 +1,82 @@
+<?php
+namespace Apple_Exporter\Components;
+
+/**
+ * An HTML's blockquote representation.
+ *
+ * @since 0.2.0
+ */
+class Pre extends Component {
+
+  /**
+   * Look for node matches for this component.
+   *
+   * @param DomNode $node
+   * @return mixed
+   * @static
+   * @access public
+   */
+  public static function node_matches( $node ) {
+    if ( 'pre' == $node->nodeName ) {
+      return $node;
+    }
+
+    return null;
+  }
+
+  /**
+   * Build the component.
+   *
+   * @param string $text
+   * @access protected
+   */
+  protected function build( $text ) {
+    preg_match( '#<pre.*?>(.*?)</pre>#si', $text, $matches );
+    $text = $matches[1];
+
+    $this->json = array(
+      'role' => 'container',
+      'components' => array( array(
+        'role'   => 'body',
+        'text'   => $this->markdown->parse( $text ),
```

Comment:
I think I see what you mean here. Like, the $text is just a just a chunk of code it doesn't need to be sent through Markdown, and in fact doing so might mess up it's formatting. 

Example as-is:

```json
{
      "role": "container",
      "components": [
        {
          "role": "body",
          "text": "$(document).ready(function () {\r\n $( '.select2' ).select2();\r\n appleNewsSettingsSortInit( '#meta-component-order-sort', 'meta_component_order' );\r\n});",
          "format": "markdown",
          "layout": "pre-layout",
          "textStyle": "default-pre"
        }
      ],
      "textStyle": "default-pre"
    },
```

Example with it removed:

```json
    {
      "role": "container",
      "components": [
        {
          "role": "body",
          "text": "$(document).ready(function () {&#13;\n $( '.select2' ).select2();&#13;\n appleNewsSettingsSortInit( '#meta-component-order-sort', 'meta_component_order' );&#13;\n});",
          "layout": "pre-layout",
          "textStyle": "default-pre"
        }
      ],
      "textStyle": "default-pre"
    },
```

Yes I think it's better... less risk of encoding characters again that shouldn't be and such.